inc/plugin.class.php

```diff
@@ -96,18 +98,20 @@ function getFromDBbyDir($dir) {
    function init() {
       global $DB, $GLPI_CACHE;
 
+      // Reset cached values (to fix potential cache corruption)
       $GLPI_CACHE->set('plugins', []);
 
       if (!isset($DB) || !$DB->connected) {
          // Cannot init plugins list if DB is not connected
          return;
       }
 
-      $this->checkStates();
+      // Fill active plugins with database state then check state of each plugin to
+      // remove thoose that cannot stay active.
       $plugins = $this->find(['state' => self::ACTIVATED]);
-
-      foreach ($plugins as $ID => $plug) {
-         $this->setLoaded($ID, $plug['directory']);
+      $GLPI_CACHE->set('plugins', array_column($plugins, 'directory', 'id'));
```

Comment:
I changed the logic here, to be sure that if, for any reason, something retrieve plugin list during initialization process, this list will be populated with last valid state (that was validated on previous request).